src/java.base/share/classes/java/lang/StringLatin1.java

```diff
@@ -179,6 +185,140 @@ static int compareToCI_UTF16(byte[] value, byte[] other) {
         return len1 - len2;
     }
 
+    private static int compareToFC0(byte[] value, int off, int last, byte[] other, int ooff, int olast) {
+        int[] folded1 = null;
+        int[] folded2 = null;
+        int k1 = off, k2 = ooff, fk1 = 0, fk2 = 0;
+        while ((k1 < last || folded1 != null && fk1 < folded1.length) &&
+               (k2 < olast || folded2 != null && fk2 < folded2.length)) {
+            int c1, c2;
+            if (folded1 != null && fk1 < folded1.length) {
+                c1 = folded1[fk1++];
+            } else {
+                c1 = getChar(value, k1++);
+                folded1 = CaseFolding.foldIfDefined(c1);
+                fk1 = 0;
+                if (folded1 != null) {
+                   c1 = folded1[fk1++];
+                }
+            }
+            if (folded2 != null && fk2 < folded2.length) {
+                c2 = folded2[fk2++];
+            } else {
+                c2 = getChar(other, k2++);
+                folded2 = CaseFolding.foldIfDefined(c2);
+                fk2 = 0;
+                if (folded2 != null) {
+                    c2 = folded2[fk2++];
+                }
+            }
+            if (c1 != c2) {
+                return c1 - c2;
+            }
+        }
+        if (k1 < last || folded1 != null && fk1 < folded1.length) {
+            return 1;
+        }
+        if (k2 < olast || folded2 != null && fk2 < folded2.length) {
+            return -1;
+        }
+        return 0;
+    }
+
+    static int compareToFC(byte[] value, byte[] other) {
+        int len = value.length;
```

Comment:
The might also be a good place to use ArraySupport.mismatch to find the lowest non-matching char.